docs/build.md

```diff
@@ -0,0 +1,129 @@
+# Build soci-snapshotter from source
+
+This document is helpful if you plan to contribute to the project (thanks!) or
+want to use the latest soci snapshotter/cli in the main branch.
+
+This document includes the following sections:
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Dependencies](#dependencies)
+- [Build soci-snapshotter](#build-soci-snapshotter)
+- [Test soci-snapshotter](#test-soci-snapshotter)
+- [(Optional) Contribute your change](#optional-contribute-your-change)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Dependencies
+
+soci-snapshotter has the following dependencies. Please follow the links or commands
+to install them on your machine:
+
+> We only mention the direct dependencies of the project. Some dependencies may
+> have their own dependencies (e.g., containerd depends on runc/cni). Please refer
+> to their doc for a complete installation guide (mainly containerd).
+
+- **[go](https://go.dev/doc/install) >= 1.18** - required to build the project;
+to confirm please check with `go version`.
+- **[containerd](https://github.com/containerd/containerd/blob/main/docs/getting-started.md) >= 1.4** -
+required to run soci-snapshotter; to confirm please check with `sudo ctr version`.
```

Comment:
Maybe we should explicitly mention that the running containerd **server** should be >=1.4, since the client version (ctr) doesn't affect the snapshotter inherently except maybe in rpull

Reply:
updated to `sudo containerd --version`. thinking again, we can get the info from containerd directly, thus avoid mentioning `ctr` completely, which is not a dependency of build.